src/shipchain_common/aws.py

```diff
@@ -0,0 +1,137 @@
+"""
+Copyright 2020 ShipChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import json
+import logging
+
+import requests
+from aws_requests_auth.boto_utils import BotoAWSRequestsAuth
+from django.conf import settings
+from influxdb_metrics.loader import log_metric, TimingMetric
+from rest_framework import status
+
+from .exceptions import AWSIoTError
+from .utils import DecimalEncoder
+
+LOG = logging.getLogger('python-common')
+
+
+class AWSClient:
+
+    METHOD_POST = 'post'
+    METHOD_PUT = 'put'
+    METHOD_GET = 'get'
+    METHOD_DELETE = 'delete'
+
+    RESPONSE_200_METHODS = [METHOD_PUT, METHOD_GET, METHOD_DELETE]
+
+    def _call(self, http_method, endpoint, payload=None, params=None):
+        metric_name = self._get_generic_endpoint_for_metric(http_method, endpoint)
+        self.url = f'{self.url}/{endpoint}'
```

Comment:
This line breaks the AWSClient when multiple invocations are made with the same instance.  The first time we are in `_call` the instance variable `url` is updated to include the endpoint.  The second time we are in this method, the instance variable `url` _already_ has the endpoint, but we append it to the url again anyway.

I don't think we have code that performs multiple AWS invocations on a single instance of an AWSClient, but this will cause someone headaches if they ever do.

Reply:
Good call, there is no reason to update the self.url we can just use a url for that one call.